src/uu/split/src/split.rs

```diff
@@ -20,80 +24,122 @@ use std::path::Path;
-        "SUFFIX",
-    );
-    opts.optopt(
-        "",
-        "filter",
-        "write to shell COMMAND file name is $FILE (Currently not implemented for Windows)",
-        "COMMAND",
-    );
-    opts.optopt("l", "lines", "put NUMBER lines per output file", "NUMBER");
-    opts.optflag(
-        "",
-        "verbose",
-        "print a diagnostic just before each output file is opened",
-    );
-    opts.optflag("h", "help", "display help and exit");
-    opts.optflag("V", "version", "output version information and exit");
-
-    let matches = match opts.parse(&args[1..]) {
-        Ok(m) => m,
-        Err(f) => crash!(1, "{}", f),
-    };
-
-    if matches.opt_present("h") {
-        let msg = format!(
-            "{0} {1}
-
-Usage:
+// TODO: pack OPTION_… into local module?
+static OPTION_SUFFIX_LENGTH: &str = "suffix-length";
+static DEFAULT_SUFFIX_LENGTH: usize = 2;
+static OPTION_BYTES: &str = "bytes";
+static OPTION_LINE_BYTES: &str = "line-bytes";
+static OPTION_NUMERIC_SUFFIXES: &str = "numeric-suffixes";
+static OPTION_ADDITIONAL_SUFFIX: &str = "additional-suffix";
+static OPTION_FILTER: &str = "filter";
+static OPTION_LINES: &str = "lines";
+static OPTION_VERBOSE: &str = "verbose";
+
+static ARG_INPUT: &str = "input";
+static ARG_PREFIX: &str = "prefix";
+
+fn get_usage() -> String {
+    format!("{0} [OPTION]... [INPUT [PREFIX]]", executable!())
+}
+fn get_long_usage() -> String {
+    String::from(
+        "Usage:
   {0} [OPTION]... [INPUT [PREFIX]]
 
 Output fixed-size pieces of INPUT to PREFIXaa, PREFIX ab, ...; default
 size is 1000, and default PREFIX is 'x'. With no INPUT, or when INPUT is
 -, read standard input.",
-            NAME, VERSION
-        );
-
-        println!(
-            "{}\nSIZE may have a multiplier suffix: b for 512, k for 1K, m for 1 Meg.",
-            opts.usage(&msg)
-        );
-        return 0;
-    }
+    )
+}
 
-    if matches.opt_present("V") {
-        println!("{} {}", NAME, VERSION);
-        return 0;
-    }
+pub fn uumain(args: impl uucore::Args) -> i32 {
+    let usage = get_usage();
+    let long_usage = get_long_usage();
+
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about("Create output files containing consecutive or interleaved sections of input")
+        .usage(&usage[..])
+        .after_help(&long_usage[..])
+        .arg(
+            Arg::with_name(OPTION_SUFFIX_LENGTH)
+                .short("a")
+                .long(OPTION_SUFFIX_LENGTH)
+                .takes_value(true)
+                .default_value(format!("{}", DEFAULT_SUFFIX_LENGTH))
+                .help("use suffixes of length N (default 2)"),
+        )
+        .arg(
+            Arg::with_name(OPTION_BYTES)
+                .short("b")
+                .conflicts_with(OPTION_LINES)
+                .conflicts_with(OPTION_LINE_BYTES)
+                .long(OPTION_BYTES)
+                .takes_value(true)
+                .default_value("2")
```

Comment:
does it has to be a string ?
it cannot be 2 directly ?

Reply:
I scheemed for this rather quickly and thought that it couldn't be done... do you have an example where a typed arg is used (is it called "typed arg")?

Reply:
you are correct, my bad, sorry!